### ISSUE TITLE
Rule Suggestion: Prefer Arrow Callback

### ISSUE THREAD

**Issue by ChristianMurphy** [eslint/eslint#3140]
A rule for ES6 to recommend use of arrow functions in callback

Warnings when Arrow function is preferred
``` js
func( function () {...} )
setTimeout(function () {...}, 3000);
```

Not Warnings when Arrow function is preferred
```js 
func( () => {...} )
setTimeout( () => {...}, 3000);
```

Warnings when standard es5 function is preferred
```js 
func( () => {...} )
setTimeout( () => {...}, 3000);
```

Not Warnings when standard es5 function is preferred
``` js
func( function () {...} )
setTimeout(function () {...}, 3000);
```

Rule Example
``` js
"prefer-arrow-callback": [2, "always"]
```

or rule could alternatively look like:
``` js
"callback-style": [2, "arrow"]
```

ref conversation in #3113

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by mysticatea** [comment]
I will work on this.

**Comment by mysticatea** [comment]
I will make `prefer-arrow-callback`, no options.
Do we need an opposite style rule?

**Comment by ilyavolodin** [comment]
@mysticatea I don't think so. Let's hold of on opposite style until somebody requests it.

**Comment by pigulla** [comment]
I think it would be a good idea to add an option to this to only enforce arrow functions for single expressions (i.e., like lambdas in Python).

```js
"prefer-arrow-callback": [2, "except-mutliple-statements"]
```

With that option set, these would be considered warnings:
```js
foo.forEach(function (item) { return item.toString(); });
foo.forEach(item => { item.bar(); return item.toString(); });
```
and these would be okay:
```js
foo.forEach(item => item.toString());
foo.forEach(function (item) { item.bar(); return item.toString(); });
```
